### PR TITLE
Extracts and Concentrates pt 5

### PR DESCRIPTION
**Pull request by NastyNate2612**
adds shatter, wax, distillate, distillate cartridges, and a whole bunch of steps in between that i dont suggest u ingest or consume

#### Summary
SUMMARY: Content "pt 5 of my extracts and concentrates mod adding into mainline"

#### Describe the solution
-shatter : form of cannabis concentrate/extract
-wax: form of cannabis extract
-distillate: form of cannabis extract
-distillate carts: used with the dab pen, also a form of cannabis extract
-and a whole bunch of comestibles that are the steps in between, i dont suggest anyone try to consume those
Most of these items wont make sense until you're going through the steps and craft the end resulting items yourself.

#### Testing
crafted and test smoked/ingested all comestibles

### CODE REVIEW

**Comment by ashGlaw** [comment]
Soon we’re gonna hafta start adding different strains of cannabis. At least maybe an indica/sativa distinction

**Comment by NastyNate2612** [comment]
Yes that's why I made the Id names the way they are, to leave room open for different concentrates from different strains

**Comment by GhostEcho** [comment]
I mean honestly, after smoking for years, while there is some distinction in effects at first, they really start blending together and just all start feeling like hybrids.

**Comment by NastyNate2612** [comment]
> 
> 
> I mean honestly, after smoking for years, while there is some distinction in effects at first, they really start blending together and just all start feeling like hybrids.

Aint that the truth... smh... the different effects with strains i guess would be to give you some incentive to grow gorilla glue#4 or Sour Diesel compared to a higher yielding strain like blue dream. OHHHH SNAP I GOT IT! Ok so what makes these strains differ irl? 
1. Value (some strains are more expensive than others due to supply and demand),
2. Yield when harvested (some strains yield exponentially more than others)
3. Amount of Cannabinoids/THC (some strains are stronger than others, and when processed yield higher quantities of crude oil)
sooooo there we go, now i have a template for making new strains when i start overhauling the farming system in the game

File: data/json/items/comestibles/med.json [hunk 1]
```diff
@@ -545,6 +545,390 @@
...
+    "color": "green",
+    "quench": -5,
+    "stim": -3,
+    "healthy": 0,
+    "fun": 15,
+    "addiction_potential": 5,
+    "flags": [ "NO_INGEST" ],
+    "use_action": {
+      "type": "consume_drug",
+      "activation_message": "You hit your dap pen.  Sooooo smooth...",
```
Comment (kevingranade) [inline data/json/items/comestibles/med.json:917]:
```suggestion
      "activation_message": "You hit your dab pen.  Sooooo smooth...",
```

File: data/json/items/comestibles/med.json [hunk 2]
```diff
@@ -545,6 +545,390 @@
...
+    "type": "COMESTIBLE",
+    "comestible_type": "MED",
+    "name": { "str": "high quality distillate (heads)", "str_pl": "high quality distillate (heads)" },
+    "description": "Concentrated cannabis extract.  This is the 'heads' portion from a distilled batch of crude oil.  It still needs to be ran through a lab grade still one more time before its a finished product.",
+    "looks_like": "honey_gold",
+    "weight": "1 g",
+    "volume": "1 ml",
+    "stack_size": 10,
+    "charges_per_use": 1,
+    "max_charges": 10,
```
Comment (kevingranade) [inline data/json/items/comestibles/med.json:827]:
```suggestion
    "charges": 10,
```

File: data/json/items/comestibles/med.json [hunk 3]
```diff
@@ -545,6 +545,390 @@
...
+    "type": "COMESTIBLE",
+    "comestible_type": "MED",
+    "name": { "str": "high quality distillate (tails)", "str_pl": "high quality distillate (tails)" },
+    "description": "Concentrated cannabis extract.  This material is the 'tails' portion from a distilled batch of crude oil.  It needs to be ran through a lab grade still another time.",
+    "looks_like": "honey_gold",
+    "weight": "1 g",
+    "volume": "1 ml",
+    "stack_size": 10,
+    "charges_per_use": 1,
+    "max_charges": 10,
```
Comment (kevingranade) [inline data/json/items/comestibles/med.json:847]:
```suggestion
    "charges": 10,
```

File: data/json/items/comestibles/med.json [hunk 4]
```diff
@@ -545,6 +545,390 @@
...
+    "type": "COMESTIBLE",
+    "comestible_type": "MED",
+    "name": { "str": "high quality distillate", "str_pl": "high quality distillate" },
+    "description": "Concentrated cannabis extract.  This style of extract goes through the most processing, eliminating most if not all harmful ingredients.  Normal cannabis contains around 20%THC, this distillate should be around 88%THC.  Can be smoked in a bowl, used in a dab pen, or used to make distillate cartridges.",
+    "looks_like": "honey_gold",
+    "weight": "1 g",
+    "volume": "1 ml",
+    "stack_size": 10,
+    "charges_per_use": 1,
+    "max_charges": 10,
```
Comment (kevingranade) [inline data/json/items/comestibles/med.json:867]:
```suggestion
    "charges": 10,
```

File: data/json/items/comestibles/med.json [hunk 5]
```diff
@@ -545,6 +545,390 @@
...
+    "price": 1000,
+    "container": "jar_3l_glass",
+    "symbol": "!",
+    "color": "green",
+    "quench": -5,
+    "stim": -3,
+    "healthy": 0,
```
Comment (kevingranade) [inline data/json/items/comestibles/med.json:874]:
```suggestion
```

File: data/json/items/comestibles/med.json [hunk 6]
```diff
@@ -545,6 +545,390 @@
...
+    "looks_like": "canister_empty",
+    "weight": "1 g",
+    "volume": "1 ml",
+    "stack_size": 50,
+    "price": 5000,
+    "symbol": "!",
+    "color": "green",
+    "quench": -5,
+    "stim": -3,
+    "healthy": 0,
```
Comment (kevingranade) [inline data/json/items/comestibles/med.json:911]:
```suggestion
```

File: data/json/items/comestibles/med.json [hunk 7]
```diff
@@ -545,6 +545,390 @@
...
+      "effects": [
+        { "id": "weed_high", "duration": "90 m" },
+        { "id": "pkill1", "duration": "90 m" },
+        { "id": "pkill2", "duration": "5 m" },
+        { "id": "high", "duration": "45 m" },
+        { "id": "smoke", "duration": "1 m" }
+      ],
+      "stat_adjustments": { "hunger": 3, "thirst": 5 },
+      "fields_produced": { "fd_weedsmoke": 2 },
+      "need_charges": 1,
```
Comment (kevingranade) [inline data/json/items/comestibles/med.json:927]:
```suggestion
      "charges_needed": { "fire": 1 },
```

Comment (NastyNate2612) [reply data/json/items/comestibles/med.json]: This particular comestible uses the Dab Pen (On) as the tool which runs on battery power so a fire source shouldn't be required.

Comment (NastyNate2612) [reply data/json/items/comestibles/med.json]: ^

**Comment by kevingranade** [comment]
Ahh I see my mistake, I had assumed it was a typo since "need_charges": 1 isn't valid.
Just need to double check that it consumes charges and such as expected and we're good to go.

**Comment by NastyNate2612** [comment]
> 
> 
> Ahh I see my mistake, I had assumed it was a typo since "need_charges": 1 isn't valid.
> Just need to double check that it consumes charges and such as expected and we're good to go.

Yeah this was the only way I could get the item to use charges properly with only doing JSON stuff. It would need a item use action similar to the e-cig if we wanted it to be perfect

**Comment by NastyNate2612** [comment]
i kinda had a similar issue with the extractors too, hence the (On) versions of them all to use power properly. Ill do some experimentation to see if i can get a tool to use charges properly when only referenced by qualities

**Comment by NastyNate2612** [comment]
Thank You Kevin for your help on this and everything you do for the game! also thank you for string cleanup, and thank you for adding my contributions :-)

**Pull request merged**